src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLanguageServerClient.cs

```diff
@@ -145,11 +162,47 @@ internal class RazorLanguageServerClient : ILanguageClient, ILanguageClientCusto
         var logHubLogger = _loggerProvider.CreateLogger("Razor");
         var razorLogger = new LoggerAdapter(logHubLogger);
         _server = RazorLanguageServerWrapper.Create(serverStream, serverStream, razorLogger, _projectSnapshotManagerDispatcher, ConfigureLanguageServer, _languageServerFeatureOptions);
-
+        await EnsureContainedLanguageServersInitializedAsync();
```

Comment:
Do we need to wait for this, or can we fire and forget so we don't regress any startup times?

Reply:
We need to ensure that `initialize` calls on dependent LanguageServers are completed before we make [other requests against them](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#initialize) or we'll get error `-32002`. An alternative would be to fire-and-forget here like you suggest and then setup some system to await full initialization when we're about to do our first delegation. that would technically allow us to get farther (out of `Activate`) before blocking, but for all the interesting scenarios I can think of we'd still be waiting for these initializes to finish, and it would sort of randomize which Handler would get the big slow request, which could be very confusing for telemetry.

Reply:
I completely agree with you, its just frustrating not kwowing if this will have a perf hit. I guess we might just have to take the potential hit to initialization, and then try to pull back some perf afterwards.